### PR TITLE
Deprecate --rocksdb-shred-compaction fifo

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
Rocksdb fifo compaction was originally added to decrease I/O and mitigate write stalls that we were encountering with level compaction. Since the introduction of fifo, the level compaction mode has been optimized to significantly reduce I/O amplification (see PR list below). With the reduced I/O, level is comparable and simplifies some other aspects (such as having to budget GB for each column with fifo).

#### Summary of Changes
Add a deprecation warning when the user sets `--rocksdb-shred-compaction fifo`; this will later be completely disallowed

Related PR's:
- https://github.com/solana-labs/solana/pull/26651
- https://github.com/solana-labs/solana/pull/27571
  - https://github.com/solana-labs/solana/pull/32548 (addressed a minor issue with 27571)
- https://github.com/solana-labs/solana/pull/29239

### CODE REVIEW

**Comment by steviez** [comment]
Two items:
- I would like this change to land in v2.0. I realize I just missed the branch cut; however, we have been communicating that this was coming. Once such example [here](https://github.com/solana-labs/solana/issues/35110#issuecomment-2046219742) from April
- I think this change should get mentioned in the changelog, regardless of which branch it makes it to

Given the two above items, I would like to mention this item in the v2.0 changelog section. But, I would also like some buy-in for the v2.0 backport before doing that. So, I will first get a "normal" review for the content, and then another review from one of the BP folks

File: validator/src/main.rs [hunk 1]
```diff
@@ -1772,16 +1772,26 @@ pub fn main() {
...
-                        matches,
-                        "rocksdb_fifo_shred_storage_size",
-                        u64
-                    ))),
-                },
+                "fifo" => {
+                    warn!(
+                        "The value \"fifo\" for --rocksdb-shred-compaction has been deprecated. \
+                         Use of \"fifo\" will still work for now, but will be completely removed \
+                         in the future. To update, use \"level\" for --rocksdb-shred-compaction, \
```
Comment (bw-solana) [inline validator/src/main.rs]: If we know which release we will remove, it might be nice to mention it here

**Review by bw-solana — DISMISSED** [review]
LGTM

File: validator/src/main.rs (hunk 1 above)
Comment (steviez) [reply validator/src/main.rs]: Fair enough. In the same vein of wanting confirmation about the BP for the change log, I would also want confirmation about the BP before putting specific numbers down.

Ideal would be to have deprecation warning show in v2.0.0, and hard error in v2.1.0

**Comment by steviez** [comment]
@CriesofCarrots - Adding you as a reviewer for initial thoughts about BP'ing this to v2.0.0

**Review by CriesofCarrots — DISMISSED** [review]
v2.0 backport seems appropriate to me

File: validator/src/main.rs (hunk 1 above)
Comment (steviez) [reply validator/src/main.rs]: Updated to state intention to remove in v2.1

**Review by CriesofCarrots — APPROVED** [review]

**Review by bw-solana — APPROVED** [review]

**Pull request merged**